collection-matchers/src/main/java/ru/yandex/qatools/matchers/collection/Wrapper.java

```diff
@@ -37,18 +37,21 @@ public int hashCode() {
     @Override
     @SuppressWarnings("unchecked")
     public boolean equals(Object obj) {
-        Wrapper<T> comparedWrapper = (Wrapper<T>) obj;
-        T toCompare = comparedWrapper.getWrapped();
+        if (obj instanceof Wrapper) {
```

Comment:
> think it should be guard method from guava or commons-lang3, as it error and should throw ex.

I propose to keep it and to follow the official javadoc: http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html#equals(java.lang.Object)

> Also please add a test

Sorry, I forgot to add them - PR was updated.